app/helpers/dag_json_helper.rb

```diff
@@ -0,0 +1,119 @@
+module DagJsonHelper
+  DAG_NAME_HOST_FILTER = "host_filter".freeze
+  DAG_NAME_ALIGNMENT = "non_host_alignment".freeze
+  DAG_NAME_POST_PROCESS = "postprocess".freeze
+  DAG_NAME_EXPERIMENTAL = "experimental".freeze
+
+  def generate_dag_json(pipeline_run, dag_name)
+    case dag_name
+    when DAG_NAME_HOST_FILTER
+      attribute_dict = host_filtering_dag_attributes(pipeline_run)
+    when DAG_NAME_ALIGNMENT
+      attribute_dict = alignment_dag_attributes(pipeline_run)
+    when DAG_NAME_POST_PROCESS
+      attribute_dict = postprocess_dag_attributes(pipeline_run)
+    when DAG_NAME_EXPERIMENTAL
+      attribute_dict = experimental_dag_attributes(pipeline_run)
+    end
+
+    sample = pipeline_run.sample
+    attribute_dict[:bucket] = SAMPLES_BUCKET_NAME
```

Comment:
Not sure where this constant  `SAMPLES_BUCKET_NAME` is coming from.

I would move this code to each method below, to avoid mutating the attribute dict here.